src/Root.js

```diff
@@ -104,13 +128,38 @@ class Root extends React.Component {
       (window &&
         window.chrome &&
         window.chrome.runtime &&
-        typeof window.chrome.runtime.getManifest === "function" &&
+        typeof window.chrome.runtime.getManifest === 'function' &&
         (window.chrome.runtime.getManifest() || {}).version) ||
       null
     );
   }
 
+  isValidChromeRuntime() {
+    return (
+      window &&
+      window.chrome &&
+      window.chrome.runtime &&
+      typeof window.chrome.runtime.getManifest === 'function'
+    );
+  }
+
   render() {
+    const {
+      hasWallet,
+      canUse,
+      shouldConfirm,
+      snaxDomain,
+      loading,
+      preparingTransaction,
+    } = this.state;
+
+    const redirectToPassword = hasWallet && !canUse;
+    const redirectToWallet = canUse && !shouldConfirm;
+    const redirectToConfirmPhrase = canUse && shouldConfirm;
+
+    if (loading) return null;
```

Comment:
@bee-warrior you need to place any content on the App, otherwise, extension window collapses to the minimum content size or 0x0.

Reply:
@pinknumbat 
I've placed 2 loaders inside the App. Can you change `<div id="loading" />` and `<div id="preparing transaction" />` to overlay loaders?